api/router.go

```diff
@@ -17,42 +16,35 @@ type apiRouter struct {
 	logger *zap.Logger
 }
 
-func readJSONRequest(r *http.Request, logger *zap.Logger) (*[]byte, *CreateCredentialRequest, error) {
+func (ar *apiRouter) readJSONRequest(r *http.Request) (*CreateCredentialRequest, error) {
 	out := new(CreateCredentialRequest)
 
 	// Validate the request body
 	if err := validateJSONRequest(r, out); err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	logger.Info("Got valid request",
+	ar.logger.Info("Got valid request",
 		zap.String("endpoint", r.URL.Path),
-		zap.String("address", out.Address),
-		zap.String("msg", out.Msg),
-		zap.String("sig", out.Sig),
+		zap.String("address", out.Address.Hex()),
+		zap.String("msg", string(out.Msg)),
+		zap.String("sig", string(out.Sig)),
```

Comment:
out.sig is a []byte that represents binary, so we need to hex-encode before logging.

Msg is expected to be valid ascii, so the cast is fine there.